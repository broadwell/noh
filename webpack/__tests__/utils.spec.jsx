import {
  getTime,
  convertTimeToSeconds,
  convertSecondsToHhmmss,
  fillGrid
} from "../utils";
import phrases from "./__fixtures__/phrases.json";

describe("getTime", () => {
  test("it converts time expressed in hh:mm:ss.mmm to seconds.millis", () => {
    const element = {};
    element.getAttribute = () => "01:01:01.001";
    const expectedTime = 3661.001;
    expect(getTime(element)).toEqual(expectedTime);
  });

  test("it defaults to 0 for invalid times", () => {
    const element = {};
    element.getAttribute = () => "01:01:90.001";
    expect(getTime(element)).toEqual(0);
  });

  test("it defaults to 0 for bad formatted times", () => {
    const element = {};
    element.getAttribute = () => "Dec 1st 2017 01:01:90.001";
    expect(getTime(element)).toEqual(0);
  });
});

describe("convertTimeToSeconds", () => {
  test("it correctly converts time expressed in hh:mm:ss to seconds", () => {
    const time = "01:01:01";
    const expectedTime = 3661;
    expect(convertTimeToSeconds(time)).toEqual(expectedTime);
  });

  test("it defaults to 0 for invalid times", () => {
    const time = "01:01:90.001";
    expect(convertTimeToSeconds(time)).toEqual(0);
  });

  test("it defaults to 0 for bad formatted times", () => {
    const time = "Dec 1st 2017 01:01:90.001";
    expect(convertTimeToSeconds(time)).toEqual(0);
  });
});

describe("convertSecondsToHhmmss", () => {
  test("it correctly converts time in seconds to hh:mm:ss", () => {
    const time = 3600;
    expect(convertSecondsToHhmmss(time)).toBe("01:00:00");
  });
});

describe("fillGrid", () => {
  test("it correctly fills in the array of data with data starting at position 2", () => {
    const grid = phrases.phrases[0].nohkan.grid; // eslint-disable-line prefer-destructuring
    const fullData = fillGrid(grid, 13);
    expect(fullData).toMatchSnapshot();
  });
  test("it correctly fills in the array of data with data starting past 2", () => {
    const grid = phrases.phrases[1].nohkan.grid; // eslint-disable-line prefer-destructuring
    const fullData = fillGrid(grid, 13);
    expect(fullData).toMatchSnapshot();
  });
});
