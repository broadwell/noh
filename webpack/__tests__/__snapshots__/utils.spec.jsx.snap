// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fillGrid it correctly fills in the array of data with data starting at position 2 1`] = `
Array [
  Object {
    "length": 1,
    "start": 0,
    "text": "",
  },
  Object {
    "length": 2,
    "start": 1,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 1,
    "start": 3,
    "text": "",
  },
  Object {
    "length": 1,
    "start": 4,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 2,
    "start": 5,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 3,
    "start": 7,
    "text": "",
  },
  Object {
    "length": 2,
    "start": 10,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 1,
    "start": 13,
    "text": "",
  },
]
`;

exports[`fillGrid it correctly fills in the array of data with data starting past 2 1`] = `
Array [
  Object {
    "length": 2,
    "start": 0,
    "text": "",
  },
  Object {
    "length": 1,
    "start": 2,
    "text": "",
  },
  Object {
    "length": 2,
    "start": 3,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 3,
    "start": 5,
    "text": "",
  },
  Object {
    "length": 1,
    "start": 8,
    "style": "normal",
    "text": "Naka no takane",
    "voices": Array [],
  },
  Object {
    "length": 4,
    "start": 10,
    "text": "",
  },
]
`;
